README.md

```diff
@@ -1,27 +1,196 @@
+import type { Signer } from "ethers";
+
+const ENCODING = "base64"; // Use base64 encoding for a clean text output
+
+export const generateAndPrintKeys = async (signer: Signer) => {
+  const clientOptions = {
+    env: "production", // this works even if in .env, XMTP_ENV=dev. if set this to dev, key bundle doesn't work.
+  };
+
+  const address = await signer.getAddress();
+  let keys: Uint8Array | null = null;
+
+  keys = await Client.getKeys(signer, {
+    ...clientOptions,
+    skipContactPublishing: true,
+    persistConversations: false,
+  });
+
+  // Output the keys in Base64 format for .env storage
+  console.log(`XMTP_KEY_BUNDLE=${Buffer.from(keys).toString(ENCODING)}`);
+
+  // Example of how you might use these keys in creating the client
+  const client = await Client.create(null, {
+    ...clientOptions,
+    privateKeyOverride: keys,
+  });
+
+  return client;
+};
+
+// Example usage
+const main = async () => {
+  const privateKey = "<YOUR_PRIVATE_KEY>"; // Replace with your private key
+  const wallet = new Wallet(privateKey);
+
+  console.log(`Using Wallet Address: ${wallet.address}`);
+
+  await generateAndPrintKeys(wallet);
+};
+
+main().catch(console.error);
 ```
 
-This project was created using `bun init` in bun v1.0.25. [Bun](https://bun.sh) is a fast all-in-one JavaScript runtime.
-## Example Env
-An example Env File could look something like:
+### Run the script
+
+Run the script to get the key bundle. Make a note of the key bundle. You’ll need it in the next step.
+
+```bash
+tsx generateAndPrintKeys.ts
+```
+
+## Set up the environment
+
+In the `broadcast-example-api` directory, create an `.env` file with the following contents, for example.
+
+Replace `<your hex string of broadcast key bundle>` with the key bundle you got from the `generateAndPrintKeys.ts` script.
+
 ```
 XMTP_ENV=dev
 PORT=6989
 
-# XMTP Broadcast specific configuration
+# XMTP Broadcast-specific configuration
 XMTP_RATE_LIMIT_AMOUNT=1000
 XMTP_RATE_LIMIT_DURATION=300000
-XMTP_KEY_BUNDLE=<hex string of broadcast keybundle>
+XMTP_KEY_BUNDLE=<your hex string of broadcast key bundle>
 XMTP_FILE_PERSISTENCE_PATH=/tmp/xmtp
 ```
+
+### Update the CORS policy
+
+In `index.ts` at the root of the `broadcast-example-api` directory, add `"http://localhost:3000"` as shown in the following example.
+
+This addition enables the broadcast API to accept requests from `http://localhost:3000` , which will be the URL of the example subscribe-broadcast page that you’ll use with the broadcast API in an upcoming step.
+
+```tsx
+app.use(
+  cors({
+    origin: ["http://localhost:3000", "https://subscribe-broadcast.vercel.app"],
+  })
+);
+```
+
+## Install dotenv
+
+If not already installed, install dotenv.
+
+```bash
+yarn add dotenv
+```
+
+### Load environment variables to your app
+
+At the top of the `src/index.ts` file, add the following line. This will load all variables from your `.env` file into `process.env`.
+
+```tsx
+import "dotenv/config";
```

Comment:
I added this to the file to make things work...but there is an existing line: `import dotenv from "dotenv";`

The commands weren't working for me without `import "dotenv/config";`.

Not sure if both are necessary? Could also be user error on my part!